v3/src/components/common/formula-insert-values-menu.tsx

```diff
@@ -25,46 +26,46 @@ export const InsertValuesMenu = ({setShowValuesMenu}: IProps) => {
   )
   const attributeNames = dataSet?.attributes.map(attr => attr.name)
   const globalManager = dataSet && getGlobalValueManager(getSharedModelManager(dataSet))
-  const globals = globalManager
-                    ? Array.from(globalManager.globals.values()).map(global => ({ label: global.name }))
-                    : []
+  const globals = useMemo(() => globalManager
+    ? Array.from(globalManager.globals.values()).map(global => ({ label: global.name }))
```

Comment:
If a new global is added to the globalManager I think the `globals` memo will not be updated. That seems like a bug?

Reply:
Good question -- I'll look into it.